test/jdk/sun/tools/jstatd/JstatGCUtilParser.java

```
@@ -43,9 +43,9 @@ public enum GcStatisticsType {
     }
 
     public enum GcStatistics {
-        S0(GcStatisticsType.PERCENTAGE),
-        S1(GcStatisticsType.PERCENTAGE),
-        E(GcStatisticsType.PERCENTAGE),
+        S0(GcStatisticsType.PERCENTAGE_OR_DASH),
+        S1(GcStatisticsType.PERCENTAGE_OR_DASH),
+        E(GcStatisticsType.PERCENTAGE_OR_DASH),
```

I don't see jstat tests in the zgc problem list. Is this fixing known test failures?

---

Yes, ZGC reports dash for eden and survivor spaces.

---

So shouldn't we have ZGC test failures then?

---

No, test passes with ZGC.

---

Before my changes test just silently ignored any GC setting and always use G1.

---

Ah, ok. That makes sense now.